actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -12,7 +12,7 @@ module CollectionCaching # :nodoc:
 
     private
       def cache_collection_render(instrumentation_payload, view, template)
-        return yield unless @options[:cached]
+        return yield unless @options[:cached] && (!view.controller.respond_to?(:perform_caching) || view.controller.perform_caching)
```

Comment:
If the controller doesn't respond to `perform_caching`, we _shouldn't_ cache, to match the other method's behaviour:

https://github.com/rails/rails/blob/9fc01836994176ada283ac3d7afb91d2ac52c555/actionview/lib/action_view/helpers/cache_helper.rb#L167

```suggestion
        return yield unless @options[:cached] && view.controller.respond_to?(:perform_caching) && view.controller.perform_caching
```

Reply:
Makes sense @eugeneius , thanks